src/diffusers/models/unets/unet_2d_blocks.py

```diff
@@ -1042,10 +1067,32 @@ def forward(
         output_states = ()
 
         for resnet, attn in zip(self.resnets, self.attentions):
-            cross_attention_kwargs.update({"scale": lora_scale})
-            hidden_states = resnet(hidden_states, temb, scale=lora_scale)
-            hidden_states = attn(hidden_states, **cross_attention_kwargs)
-            output_states = output_states + (hidden_states,)
+            if self.training and self.gradient_checkpointing:
+
+                def create_custom_forward(module, return_dict=None):
+                    def custom_forward(*inputs):
+                        if return_dict is not None:
+                            return module(*inputs, return_dict=return_dict)
+                        else:
+                            return module(*inputs)
+
+                    return custom_forward
```

Comment:
would something like this work?
```suggestion
                def create_custom_forward(module, scale=None):
                    def custom_forward(*inputs):
                        if scale is not None:
                            return module(*inputs, scale=lora_scale)
                        else:
                            return module(*inputs)

                    return custom_forward
```

Reply:
My current understanding is that the `scale` argument should only be supplied at inference time, not both at training time and inference time as I originally thought. Since `custom_forward` is only used when gradient checkpointing during training, I'm don't think it should be called with a non-default `scale` argument.

For this PR, I would like to keep the `UNet2DModel`/`AttnDownBlock2D`/... gradient checkpointing logic as close to the current gradient checkpointing logic in `UNet2DConditionModel`/`CrossAttnDownBlock2D`/..., following https://github.com/huggingface/diffusers/pull/6718#issuecomment-1913574196 and https://github.com/huggingface/diffusers/pull/6718#issuecomment-1913846617, and implement general changes to the `lora_scale` handling logic in #6756 and any follow up PRs.

Reply:
ok :)